### DISCUSSION TITLE
Scripts and Branching jobs lock up the RQWorker?

### DISCUSSION THREAD

**Discussion by jchambers2012** [netboxlabs/netbox-branching#134] [Q&A]
Based on some simple testing the way the job schedule and branching seams to work today the active script running blocks the create a new branch job when creating the new branch via the UI. Is this correct?

As custom script support becomes available, will scripts be able to create branches as part of its job run?  Let’s say I have a script that is designed to build out a new location with lots of IPAMs and Devices.  Will the script be able to create a new branch and wait for it to be created before adding the new objects? 

From a UX POV, it would be nice to allow custom scripts to have the ability to create new branches so the users will not need to see a AbortScript error to make sure it is in branch or a branch with a particular name; like new-site-SiteXYZ. Maybe the script subsystem needs to add support to “yield” or suspend the current job to another job so the branch is created and ready?

**Comment by FloSch62** [comment]
I am also very interested into that... 
Especially for me its important to apply the changes of a script to a branch instead to main
